sources/console.c

```diff
@@ -191,7 +305,93 @@ od_console_show_stats(od_client_t *client, machine_msg_t *stream)
 	od_route_pool_stat_database(
 	  &router->route_pool, od_console_show_stats_cb, cron->stat_time_us, argv);
 
-	return kiwi_be_write_complete(stream, "SHOW", 5);
+	int rc = kiwi_be_write_complete(stream, "SHOW", 5);
+	if (rc == -1) {
+		return rc;
+	}
+
+	return rc;
+}
+
+static inline od_retcode_t
+od_console_show_errors_frontend(od_client_t *client, machine_msg_t *stream)
+{
+	assert(stream);
+	od_router_t *router = client->global->router;
+
+	void *argv[] = { stream };
+
+	machine_msg_t *msg;
+	msg = kiwi_be_write_row_descriptionf(
+	  stream, "sssl", "database", "user", "error_type", "count");
+
+	if (msg == NULL)
+		return OD_GENERAL_NOTOK_RESPONSE_CODE;
+
+	int rc;
+	rc = od_route_pool_stat_err_frontend(
+	  &router->route_pool, od_console_show_err_frontend_stats_cb, argv);
+
+	if (rc != OD_GENERAL_OK_RESPONSE_CODE)
+		return rc;
+
+	rc = kiwi_be_write_complete(stream, "SHOW", 5);
+	return rc;
+}
+
+static inline od_retcode_t
+od_console_show_errors_router(od_client_t *client, machine_msg_t *stream)
+{
+	assert(stream);
+	od_router_t *router = client->global->router;
+
+	void *argv[] = { stream };
+
+	machine_msg_t *msg;
+	msg = kiwi_be_write_row_descriptionf(stream, "sl", "error_type", "count");
+
+	if (msg == NULL)
+		return OD_GENERAL_NOTOK_RESPONSE_CODE;
+
+	int rc;
+	rc = od_route_pool_stat_err_router(
+	  router, od_console_show_err_router_stats_cb, argv);
+
+	if (rc != OD_GENERAL_OK_RESPONSE_CODE)
+		return rc;
+
+	rc = kiwi_be_write_complete(stream, "SHOW", 5);
+	return rc;
+}
+
+static inline od_retcode_t
+od_console_show_errors(od_client_t *client,
+                       machine_msg_t *stream,
+                       od_parser_t *parser)
+{
+
+	od_token_t token;
+	int rc;
+	rc = od_parser_next(parser, &token);
+	switch (rc) {
+		case OD_PARSER_KEYWORD:
+			break;
+		case OD_PARSER_EOF:
+		default:
+			return OD_GENERAL_NOTOK_RESPONSE_CODE;
+	}
+	od_keyword_t *keyword;
+	keyword = od_keyword_match(od_console_keywords, &token);
+	if (keyword == NULL)
+		return -1;
+	switch (keyword->id) {
+		case OD_LFRONTEND:
```

Comment:
Do we really really want two different views for FE and BE errors? Maby just show them in different lines of the same view?

Reply:
merged into one